modules/navigation/navigation_mesh_generator.cpp

```diff
@@ -445,7 +445,7 @@ void NavigationMeshGenerator::_build_recast_navigation_mesh(
 	cfg.minRegionArea = (int)(p_nav_mesh->get_region_min_size() * p_nav_mesh->get_region_min_size());
 	cfg.mergeRegionArea = (int)(p_nav_mesh->get_region_merge_size() * p_nav_mesh->get_region_merge_size());
 	cfg.maxVertsPerPoly = (int)p_nav_mesh->get_verts_per_poly();
-	cfg.detailSampleDist = p_nav_mesh->get_detail_sample_distance() < 0.9f ? 0 : p_nav_mesh->get_cell_size() * p_nav_mesh->get_detail_sample_distance();
+	cfg.detailSampleDist = p_nav_mesh->get_detail_sample_distance() < 0.1f ? 0.1f : p_nav_mesh->get_cell_size() * p_nav_mesh->get_detail_sample_distance();
```

Comment:
Maybe you can instead remove this check here, and add it in `set_detail_sample_distance`? There's already a validation that it's not less than 0, it could instead validate that it's not less than 0.1f. So at least if users try to set it they get an error and not just a silent clamp.

Reply:
Does the setter also run when someone loads in an old navmesh resource with a stored zero value?
I will change the setter to 0.1 regardless but I would like to keep this line to avoid any chance to crash ReCast with old navmesh properties.

Reply:
Yeah properties always go through the setter, also when reading from config, so it should be safe.